src/main/kotlin/org/radarbase/gateway/Config.kt

```diff
@@ -28,43 +32,96 @@ data class Config(
     fun validate() {
         kafka.validate()
         auth.validate()
+        pushIntegration.validate()
+    }
+}
+
+data class PushIntegrationConfig(
+    val enabledServices: List<String> = listOf("garmin"),
+    val garmin: GarminConfig = GarminConfig()
+) {
+    fun validate() {
+        enabledServices.forEach { service ->
```

Comment:
just validate the service configs, they can see what other validation is needed.